test/test_torch.py

```diff
@@ -18395,7 +18419,7 @@ def _make_tensor(shape, dtype, device, fill_ones=False) -> torch.Tensor:
         return torch.ones(*shape, dtype=_convert_t(dtype, device), device=device)
 
     # Returns a tensor with random integer values
-    if not dtype.is_floating_point:
+    if not dtype.is_floating_point or not dtype.is_complex:
```

Comment:
`if !(dtype.is_floating_point and dtype.is_complex)`

Won't your current check always go down this codepath?